### PR TITLE
 Integration tests for application clone with rules and label selectors

### PR DESCRIPTION
**Pull request by disrani-px**
**What type of PR is this?**
Integration Test
Bug

**What this PR does / why we need it**:
Added more integration tests for application clone and fixed issues found in the tests

**Does this PR change a user-facing CRD or CLI?**:

No

**Is a release note needed?**:

No

**Does this change need to be cherry-picked to a release branch?**:

No

### CODE REVIEW

**Comment by disrani-px** [comment]
The tests are passing
```
    --- PASS: TestMain/ApplicationClone (3846.25s)
        --- PASS: TestMain/ApplicationClone/deploymentTest (214.42s)
        --- PASS: TestMain/ApplicationClone/statefulsetTest (804.16s)
        --- PASS: TestMain/ApplicationClone/statefulsetRuleTest (868.10s)
        --- PASS: TestMain/ApplicationClone/preExecRuleMissingTest (187.21s)
        --- PASS: TestMain/ApplicationClone/postExecRuleMissingTest (177.55s)
        --- PASS: TestMain/ApplicationClone/disallowedNamespaceTest (187.96s)
        --- PASS: TestMain/ApplicationClone/failingPreExecRuleTest (188.68s)
        --- PASS: TestMain/ApplicationClone/failingPostExecRuleTest (177.42s)
        --- PASS: TestMain/ApplicationClone/labelSelectorTest (1040.74s)
```

File: test/integration_test/applicationclone_test.go [hunk 1]
```diff
@@ -10,34 +10,190 @@ import (
...
+		// Make sure the cloned app is running
+		err = schedulerDriver.UpdateTasksID(cloneAppCtx, appKey+"-"+instanceID+"-dest")
+		require.NoError(t, err, "Error updating task id for app clone context")
+		err = schedulerDriver.WaitForRunning(cloneAppCtx, defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for cloned app to get to running state")
 
-	// Destroy the clone task and cloned add
-	destroyAndWait(t, []*scheduler.Context{cloneAppCtx, cloneTaskCtx[0]})
+		// Destroy the clone task and cloned app
+		err = schedulerDriver.UpdateTasksID(cloneAppCtx, appKey+"-"+instanceID+"-dest")
```
Comment (Rohit-PX) [inline test/integration_test/applicationclone_test.go]: Can we destroy without updating the taskID?

File: test/integration_test/applicationclone_test.go [hunk 2]
```diff
@@ -10,34 +10,190 @@ import (
...
+		destroyAndWait(t, []*scheduler.Context{cloneAppCtx, cloneTaskCtx[0]})
+	} else {
+		require.Error(t, err, "Expected app clone task to fail")
+		// Destroy the clone task
+		destroyAndWait(t, cloneTaskCtx)
+	}
 
 	// Destroy the original app
-	err = schedulerDriver.UpdateTasksID(ctx, ctx.GetID())
+	err = schedulerDriver.UpdateTasksID(ctxs[0], ctxs[0].GetID())
```
Comment (Rohit-PX) [inline test/integration_test/applicationclone_test.go]: Same comment as above.

File: test/integration_test/specs/mysql-clone-disallowed-ns/appclone.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: stork.libopenstorage.org/v1alpha1
+kind: ApplicationClone
+metadata:
+  name: clonemysql
```
Comment (Rohit-PX) [inline test/integration_test/specs/mysql-clone-disallowed-ns/appclone.yaml]: nit: clone-mysql
You may change everywhere if you want.

File: test/integration_test/specs/cassandra-clone/appclone.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: stork.libopenstorage.org/v1alpha1
+kind: ApplicationClone
+metadata:
+  name: clonecassandra
```
Comment (Rohit-PX) [inline test/integration_test/specs/cassandra-clone/appclone.yaml]: nit: clone-cassandra
You may change if you wish.

File: test/integration_test/applicationclone_test.go (hunk 1 above)
Comment (disrani-px) [reply test/integration_test/applicationclone_test.go]: Yeah don't need this, will remove it

File: test/integration_test/applicationclone_test.go (hunk 2 above)
Comment (disrani-px) [reply test/integration_test/applicationclone_test.go]: This is required. Unfortunately the `DeepCopy()` doesn't really do a deep copy of the app specs. So it'll be set to the cloned apps namespace when we are validating that above.

File: test/integration_test/specs/mysql-clone-disallowed-ns/appclone.yaml (hunk 3 above)
Comment (disrani-px) [reply test/integration_test/specs/mysql-clone-disallowed-ns/appclone.yaml]: Updated

File: test/integration_test/specs/cassandra-clone/appclone.yaml (hunk 4 above)
Comment (disrani-px) [reply test/integration_test/specs/cassandra-clone/appclone.yaml]: Updated

**Pull request merged**